syntax = "proto3";

package motioncontroller;

import "google/protobuf/empty.proto";

service DeviceService {
  rpc SayHello (HelloRequest) returns (HelloReply) {}
  rpc GetDigitalValues(google.protobuf.Empty) returns (Analog2Digital) {}
  rpc OutVoltage (Digital2Analog) returns (CommandResult) {}
  rpc ConfigureAnalog2Digital(Analog2DigitalConfig) returns (CommandResult) {}
  rpc SetDiffChannel(SetDiffChnAnalog2Digital) returns (CommandResult) {}
}

message HelloRequest {
  string name = 1;
}

message HelloReply {
  string message = 1;
}

message CommandResult {
  string command_result = 1;
}

message Digital2Analog {
  string  channel_name  = 1;
  float channel_voltage = 2;
}

message Analog2Digital {
  string channel_name = 1;
}

message Analog2DigitalConfig {
  int32 gain      = 1;
  int32 data_rate = 2;
}

message SetDiffChnAnalog2Digital {
  int32 channel_id = 1;
}